### PR TITLE
Add limitation in the array of events for the endpoint of generate

### PR DESCRIPTION
**Pull request by shudhansu-shekhar**
### Description of the Change

In this after making this change, limitation is added in the array of events for the endpoint of generate

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: <shudhansu shekhar> <shudhansu.shekhar.ext@ericsson.com>

### CODE REVIEW

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 1]
```diff
@@ -76,6 +76,13 @@ public class RemremGenerateController {
     @Value("${lenientValidationEnabledToUsers:false}")
     private boolean lenientValidationEnabledToUsers;
 
+    @Value("${maxEventsOfInputArray:250}")
+    private int maxSize;
```
Comment (z-sztrom) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:80]: Rename the variable to be more expressive.

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 2]
```diff
@@ -159,6 +166,14 @@ public ResponseEntity<?> generate(final String msgProtocol, final String msgType
 
             if (inputData.isJsonArray()) {
                 JsonArray inputEventJsonArray = inputData.getAsJsonArray();
+
+                if (inputEventJsonArray.size() > maxSize) {
+                    return createResponseEntity(HttpStatus.BAD_REQUEST, JSON_ERROR_STATUS,
+                            "The number of events in the input array is exceeded the allowed limit of 250 events. " +
```
Comment (z-sztrom) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:172]: Use real values, not hard-coded ones.

File: service/src/main/resources/application.properties [hunk 3]
```diff
@@ -28,4 +28,7 @@ event-repository.enabled: false
 event-repository.url: http://<host>:<port>/<context-path-if-available>
 
 # lenientValidationEnabledToUsers true will perform the validation only on mandatory fields, non-mandatory validation failures add into Eiffel message as property remremGenerateFailures
-lenientValidationEnabledToUsers: false
\ No newline at end of file
+lenientValidationEnabledToUsers: false
+
+#Maximum size of events in array is fetched from REMReM property and checked during publishing.
```
Comment (z-sztrom) [inline service/src/main/resources/application.properties:33]: Maximum number of templates in array passed to /xxx endpoint. Please, replace xxx by real endpoint name.

**Review by z-sztrom — CHANGES_REQUESTED** [review]

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java (hunk 1 above)
Comment (shudhansu-shekhar) [reply service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:80]: Done

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java (hunk 2 above)
Comment (shudhansu-shekhar) [reply service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:172]: Done

File: service/src/main/resources/application.properties (hunk 3 above)
Comment (shudhansu-shekhar) [reply service/src/main/resources/application.properties:33]: Done

**Review by shudhansu-shekhar — COMMENTED** [review]
Done, make changes as you suggest

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 4]
```diff
@@ -159,6 +162,12 @@ public ResponseEntity<?> generate(final String msgProtocol, final String msgType
 
             if (inputData.isJsonArray()) {
                 JsonArray inputEventJsonArray = inputData.getAsJsonArray();
+
+                if (inputEventJsonArray.size() > maxSizeOfInputArray) {
+                    return createResponseEntity(HttpStatus.BAD_REQUEST, JSON_ERROR_STATUS,
+                            "The number of events in the input array is to high: " + inputEventJsonArray.size() + " > " + maxSizeOfInputArray + ", " +
```
Comment (z-sztrom) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:168]: -> too high: "

Comment (z-sztrom) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:168]: -> maxSizeOfInputArray + "; " + "you can ... it."

**Review by z-sztrom — CHANGES_REQUESTED** [review]

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java (hunk 4 above)
Comment (shudhansu-shekhar) [reply service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:168]: Done

Comment (shudhansu-shekhar) [reply service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java:168]: Done

**Review by shudhansu-shekhar — COMMENTED** [review]
Done, please check latest code

**Review by z-sztrom — APPROVED** [review]

**Pull request merged**